actions/microsoft-onedrive/microsoft_onedrive/onedrive_get_action.py

```diff
@@ -0,0 +1,331 @@
+    except Exception as e:
+        raise ActionError(f"Unexpected error while listing OneDrive folders: {str(e)}")
+
+
+@action
+def list_items_from_onedrive_folder(
+    token: OAuth2Secret[
+        Literal["microsoft"],
+        List[Literal["Files.Read"]],
+    ],
+    params: OneDriveListingParams,
+) -> Response[dict]:
+    """
+    List items in a specified OneDrive folder. Use "/" for the root folder, "/Documents", etc.
+    Returns also download urls (note that it's slow to print).
+
+    Args:
+        token: OAuth2 token.
+        params: Pydantic model containing folder_path and item_type.
+
+    Returns:
+        Dictionary with an 'items' key mapping to a list of item details.
+    """
+    headers = build_headers(token)
+
+    if params.folder_path in ["/", "root", "home"]:
+        url = f"{BASE_GRAPH_URL}/me/drive/root/children"
+    else:
+        folder_path = params.folder_path.lstrip("/")
+        url = f"{BASE_GRAPH_URL}/me/drive/root:/{folder_path}:/children"
+
+    response = requests.get(url, headers=headers)
+
+    if response.status_code == 200:
+        parsed_items = parse_onedrive_items(response.json(), params.item_type)
+        return Response(result={"items": parsed_items})
+    else:
+        raise ActionError(f"Failed to list OneDrive items: {response.text}")
+
+
+@action
+def list_onedrive_items_shared_with_me(
+    token: OAuth2Secret[
+        Literal["microsoft"],
+        List[Literal["Files.Read.All"]],
+    ],
+) -> Response[dict]:
+    """
+    List items shared with me in OneDrive.
+
+    Args:
+        token: OAuth2 token to use for the operation.
+
+    Returns:
+        Response containing a dictionary with an 'items' key mapping to a list of shared item details.
+    """
+    headers = build_headers(token)
+    url = f"{BASE_GRAPH_URL}/me/drive/sharedWithMe"
+
+    response = requests.get(url, headers=headers)
+
+    if response.status_code == 200:
+        items = response.json().get("value", [])
+        return Response(result={"items": items})
+    else:
+        raise ActionError(f"Failed to list shared OneDrive items: {response.text}")
+
+
+@action
+def search_for_onedrive_items(
+    token: OAuth2Secret[
+        Literal["microsoft"],
+        List[Literal["Files.Read.All"]],
+    ],
+    search_request: OneDriveSearchItemsRequest,
+) -> Response[dict]:
+    """
+    Search for items ("all" ,"files" or "folders") in OneDrive by name or content.
+
+    Args:
+        token: OAuth2 token to use for the operation.
+        search_request: OneDriveSearchItemsRequest containing the search query and item type.
+
+    Returns:
+        Response containing a list of items matching the search query and type, including download URLs for files.
+    """
+    headers = build_headers(token)
+    url = f"{BASE_GRAPH_URL}/me/drive/root/search(q='{search_request.query}')"
+
+    response = requests.get(url, headers=headers)
+
+    if response.status_code == 200:
```

Comment:
I would just put here `response.raise_for_status()` and remove one level of indentation